### ISSUE TITLE
Multi-window multi-entry solution

### ISSUE THREAD

**Issue by nodgear** [fliegwerk/electron-react-boilerplate#180]
Fidget with the template a little to figure out the code, it seemed similar to CRA code, so i quickly added craco + tailwind.
But here's a small detail of my project: it needs multiple .html files, routing just doesn't work for me because i allow thirdparty plugins/widgets to create windows on my electron app.

~~Tried to use react-scripts eject in order to get a webpack file and add more entries to it, but it fails.~~ Managed to eject, but it also breaks everything, adding the new html/tsx files to the path will not be enough anyway.

Any good solution for my issue? i would love to use this template instead of the one i'm currently using, but keeping the multiple .html files i could get with parcel.

**Comment by fussel178** [comment]
Hi!
Thank you for using (and testing) our boilerplate! That means a lot to us! :tada: 

And now to your issue:
At the moment the project does not support multiple electron windows out of the box.

One little question regarding your setup:
Do parcel insert the built html files into the `public` folder?

I think, the trick is to extend the electron main process to allow windows can open other windows via IPC.
You need a little bit of boilerplate in the `electron` folder. I implemented it already and are happy to share if you like.
In "renderer" process you can then call `window.ipc.send` on a registered IPC channel with the url the current window should open in a new window.

For example:
```ts
// main
const pages: string[] = [
  isDev ? 'http://localhost:3000/' : `file://${__dirname}/index.html`,
  // can potentially break, because webpack dev server does not provide this file by default
  isDev ? 'http://localhost:3000/' : `file://${__dirname}/page1.html`,
  // can potentially break, because webpack dev server does not provide this file by default
  isDev ? 'http://localhost:3000/' : `file://${__dirname}/page2.html`
];

class IPCManager {
  //...
  openNewWindow(id: number): void {
    ElectronWindow.create(pages[id]);
  }
  //...
}
```

```ts
// renderer
const { send } = window.ipc;

// somewhere where you want to open a new html file
send(openNewWindow, 1); // should open page1.html
send(openNewWindow, 2); // should open page2.html
```

I hope, I can help.
Feel free to keep us up to date about any updates regarding this topic! :wink:

**Comment by nodgear** [comment]
Yes, parcel will generate files for each .html that i have, as long as i add them to my watch/build list and include the main .tsx on the html file.

the problem with parcel is that it doesn't display electron debug information, breaking a component leaves you in the dark to solve the issue, sometimes the building lifecycle also breaks, those are the reasons why i'm moving to webpack/or/babel again.

opening the multiple windows is not the issue, since i alredy know how to do that.
What i'm trying to do is to output multiple html files (currently, as react default, it only outputs to the index.html + bundle)

index.html ( react entry point: index.tsx )
page1.html (react entry point: page1.tsx )

currently everything is bundled into only index.html

**Comment by fussel178** [comment]
Okay, now I get your point.

I think, the cleanest and safeted solution is to eject and update the paths and webpack html configuration in the ejected react-scripts or write a webpack config yourself.

Short reference to Stack Overflow: :stuck_out_tongue_winking_eye: 
https://stackoverflow.com/questions/39798095/multiple-html-files-using-webpack
You can modify the `html-webpack-plugin` so that it packages different html files at the same time and specify multiple entry files.

I'm honest, ejecting was not a solution and so I never tried it with this boilerplate. :see_no_evil: 

If you want some suggestions, you can take a look at our recent project:
https://github.com/fliegwerk/trc
(shameless plug :smile:)

In short, we tried to compile TypeScript React code with webpack in simple manner and with a programmatic API. (without react-scripts and with webpack 5)
One advice, this project is currently is in a really early state and some parts might not work as expected (e.g. chunking in the build step does not work at the moment) but you can give it a try. :wink:

**Comment by nodgear** [comment]
Lovely! I will check your new project.

I tried ejecting and using the webpack html plugin approach before starting this issue, my webpack skills seems to be failing, because i can't compile anymore.
But i am willing to give the documentations one more good read before forfeit.

Thank you for your time, if nothing works i am just going to try and route stuff 😞
